pkg/storage/ingest/fetcher.go

```diff
@@ -288,7 +288,11 @@ func newConcurrentFetchers(
 	if !topics.Has(topic) {
 		return nil, fmt.Errorf("failed to find topic ID: topic not found")
 	}
-	f.topicID = topics[topic].ID
+	topicMetadata := topics[topic]
```

Comment:
I would feel more comfortable if we just check `topics.Error()`. In case of any error we just return and will be retried. WDYT?